pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,103 @@
+""" Functions to access data from the Solcast API.
+.. codeauthor:: Solcast<modelling@solcast.com.au>
+"""
+import requests
+import pandas as pd
+
+BASE_URL = "https://api.solcast.com.au/data"
+
+
+# TODO: https://pvlib-python.readthedocs.io/en/stable/user_guide/variables_style_rules.html
+PARAMETERS_MAPPING = dict(
+
+)
+
+
+def get_solcast(
+        endpoint,
+        api_key,
+        **kwargs
+):
+    """retrieves weather, irradiance and power data from the Solcast API
+    and returns it into a Pandas DataFrame.
+
+    Parameters
+    ----------
+    endpoint : str
+        one of Solcast API endpoint:
+            - live/radiation_and_weather

```

Comment:
Consider limiting the scope to radiation_and_weather only. Would allow the function to be more opinionated on the arguments rather than relying on kwargs and someone reading the API docs.